docs/DeveloperGuide.md

```diff
@@ -2,33 +2,166 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+**`HdBuy`** has [`HdBuy`](https://https://github.com/AY2021S2-CS2113-F10-1/tp/blob/master/src/main/java/seedu/hdbuy/HdBuy.java). It is responsible for,
+* At app launch: Loads previously shortlisted resale flats, if any, and starts receiving user commands.
+* At shut down: Clears search history and saves shortlisted resale flats to local file.
 
+The rest of the App consists of:
+
+* [**`Api`**](#api-component): Retrieves data on resale flats from server.
+* [**`Command`**](#command-component): The command executor.
+* [**`Parser`**](#parser-component): Translate user input to valid commands to be executed.
+* [**`Ui`**](#ui-component): Communicates with user via messages.
+* [**`Data`**](#data-component): Contains shortlist, user input history and temporarily tracks search history.
+* [**`Storage`**](#storage-component): Reads shortlisted units from, and writes shortlisted units to, the text file.
+* [**`Common`**](#common-component): Models of objects used internally.
+
+### Api component
+
+![Structure of the Api Component](diagrams/ApiClass.png)
+
+The `Api`,
+
+* receives a raw query from Find Command, query will contain filter conditions.
+* creates a connection to remote server containing data on resale flats.
+* formats raw query into valid query to be sent as a GET request.
+* updates all flats matching filter conditions in SearchedUnits class of Data component.
+
+**How classes within Api component interact with each other**
+
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where a Find command is executed.
+
+<img src="diagrams/ApiSequence.png" />
```

Comment:
![image](https://user-images.githubusercontent.com/77402245/113237036-ae4ff900-92d8-11eb-87a5-7f886a50009c.png)

Perhaps you could include a return arrow to QueryFormatter after each case in the alt block to show returning of the control.